app/assets/javascripts/discourse/app/routes/signup.js

```diff
@@ -69,29 +70,28 @@ export default class extends DiscourseRoute {
       }
     }
 
-    // When Discourse Connect is enabled, redirect to the SSO endpoint
-    if (auth_immediately && enable_discourse_connect) {
-      const returnPath = cookie("destination_url")
-        ? getURL("/")
-        : encodeURIComponent(url);
-      window.location = getURL(`/session/sso?return_path=${returnPath}`);
-      return;
-    }
-
     // Automatically kick off the external login if it's the only one available
-    if (isOnlyOneExternalLoginMethod) {
-      if (auth_immediately || login_required || !from || wantsTo) {
+    if (enable_discourse_connect) {
+      if (redirect) {
+        this.#isRedirecting = true;
+        const returnPath = cookie("destination_url")
+          ? getURL("/")
+          : encodeURIComponent(url);
+        window.location = getURL(`/session/sso?return_path=${returnPath}`);
+      } else {
+        router.replaceWith("discovery.login-required");
+      }
+    } else if (isOnlyOneExternalLoginMethod) {
+      if (redirect) {
         this.#isRedirecting = true;
-        singleExternalLogin({ signup: true });
+        singleExternalLogin();
```

Comment:
Dropping this parameter feels like an unrelated change? Not a blocker, but not sure why it's being dropped.